File: Python/7. Reverse Integer.py
class Solution(object):

    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        INT_MAX = 2**31 - 1

        sign = -1 if x < 0 else 1
        x = abs(x)
        answer = 0

        while x != 0:
            digit = x % 10
            x //= 10

            if answer > (INT_MAX - digit) // 10:
                return 0

            answer = answer * 10 + digit

        return answer * sign

File: JavaScript/7. Reverse Integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
  const INT_MAX = 2147483647;
  const INT_MIN = -2147483648;

  let answer = 0;

  while (x !== 0) {
    let digit = x % 10;
    x = Math.trunc(x / 10);

    if (
      answer > Math.trunc(INT_MAX / 10) ||
      answer < Math.trunc(INT_MIN / 10)
    ) {
      return 0;
    }

    answer = answer * 10 + digit;
  }

  return answer;
};
